egs/swbd/s5c/local/nnet3/run_cnn_tdnn.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+
+# Copyright 2015  Johns Hopkins University (Author: Daniel Povey).
+#           2015  Vijayaditya Peddinti
+#           2016  Tom Ko
+# Apache 2.0.
+
+# This is the CNN+TDNN system built in nnet3
+# This system differs from the standard TDNN system
+# in replacing the LDA with CNN at the front of the network.
+# and using fbank features as the CNN input
+# As the dimension of the CNN output is usually large, we place
+# a linear layer at the output of CNN for dimension reduction
+# The ivectors are processed through a fully connected affine layer,
+# then concatenated with the dimension-reduced CNN output and 
+# passed to the deeper part of the network.
+# Due to the data compression issue, it is better to convert MFCC
+# to FBANK in the network instead of directly using FBANK features
+# from the storage. This script uses MFCC features as its input
+# and were converted to FBANK features with an inverse of DCT matrix
+# at the first layer of the network.
+
+
+# At this script level we don't support not running on GPU, as it would be painfully slow.
+# If you want to run without GPU you'd have to call train_tdnn.sh with --gpu false,
+# --num-threads 16 and --minibatch-size 128.
+
+stage=9
+train_stage=-10
+has_fisher=true
+speed_perturb=true
+
+# CNN options
+# Parameter indices used for each CNN layer
+# Format: layer<CNN_index>/<parameter_indices>....layer<CNN_index>/<parameter_indices>
+# The <parameter_indices> for each CNN layer must contain 11 positive integers.
+# The first 5 integers correspond to the parameter of ConvolutionComponent:
+# <filt_x_dim, filt_y_dim, filt_x_step, filt_y_step, num_filters>
+# The next 6 integers correspond to the parameter of MaxpoolingComponent:
+# <pool_x_size, pool_y_size, pool_z_size, pool_x_step, pool_y_step, pool_z_step>
+cnn_indexes="3,8,1,1,256,1,3,1,1,3,1"
```

Comment:
I think there is a nice way to pass these parameters. argparse in python allows the use of same optional argument several times so you could specify the cnn-layer parameter multiple times like this.

<code>--cnn-layer "filter_x_dim=3 filt_y_dim=8" --cnn-layer "filt_x_dim=3 filt_y_dim=3" </code>

In the make_configs.py script you will declare --cnn-layer argument like this

<code>parser.add_argument('--cnn-layer', action='append') </code>

This basically specifies that if multiple cnn-layer arguments are specified you will get a list with each of these arguments when you do parser.parse_args.